core/DataTable/Filter/AddColumnsProcessedMetricsGoal.php

```diff
@@ -129,13 +146,26 @@ public function filter($table)
                 // When the table is displayed by clicking on the flag icon, we only display the columns
                 // Visits, Conversions, Per goal conversion rate, Revenue
                 if ($this->processOnlyIdGoal == self::GOALS_OVERVIEW) {
-                    continue;
+                   continue;
                 }
 
                 $extraProcessedMetrics[] = new Conversions($idSite, $idGoal); // PerGoal\Conversions or GoalSpecific\
                 $extraProcessedMetrics[] = new GoalSpecificRevenuePerVisit($idSite, $idGoal); // PerGoal\Revenue
                 $extraProcessedMetrics[] = new Revenue($idSite, $idGoal); // PerGoal\Revenue
 
+                if ($this->processOnlyIdGoal == self::GOALS_PAGES) {
+                    $extraProcessedMetrics[] = new ConversionsAttrib($idSite, $idGoal); // PerGoal\Conversions or GoalSpecific\
+                    $extraProcessedMetrics[] = new RevenueAttrib($idSite, $idGoal); // PerGoal\Revenue attrib
+                    $extraProcessedMetrics[] = new ConversionPageRate($idSite, $idGoal); // PerGoal\ConversionRate for page uniq views
+                }
+
+                if ($this->processOnlyIdGoal == self::GOALS_ENTRY_PAGES) {
+                    $extraProcessedMetrics[] = new ConversionsEntry($idSite, $idGoal); // PerGoal\Conversions or GoalSpecific\
+                    $extraProcessedMetrics[] = new GoalSpecificRevenuePerEntry($idSite, $idGoal); // PerGoal\Revenue entries
+                    $extraProcessedMetrics[] = new RevenueEntry($idSite, $idGoal); // PerGoal\Revenue entrances
+                    $extraProcessedMetrics[] = new ConversionEntryRate($idSite, $idGoal); // PerGoal\ConversionRate for entrances
+                }
+
                 if ($this->isEcommerce) {
                     $extraProcessedMetrics[] = new AverageOrderRevenue($idSite, $idGoal);
                     $extraProcessedMetrics[] = new ItemsCount($idSite, $idGoal);
```

Comment:
Those columns are currently shown on the `Sales per pages` reports. I guess they are currently not calculated. So either we need to add the columns to the aggregating or don't show them in that case.

Reply:
The new `AddColumnsProcessedMetricsGoal` constant set in the visualization was overwriting the ecommerce constant so these metrics were not added. Not sure if it's the nicest solution, but I've added extra two `AddColumnsProcessedMetricsGoal` constants for the page / entry page goals with ecommerce so that this combination is supported. This allows the ecommerce metrics to be correctly added (if ecommerce is enabled) when viewing the page goal metrics.